packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { SwitchProps, SwitchInfo } from './Switch.types';
+import { LayoutAnimation } from 'react-native';
+import { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+
+const defaultAccessibilityActions = [{ name: 'Toggle' }];
+
+export const useSwitch = (props: SwitchProps): SwitchInfo => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    onChange,
+    checked,
+    defaultChecked,
+    label,
+    labelPosition,
+    componentRef = defaultComponentRef,
+    disabled,
+    accessibilityRole,
+    ...rest
+  } = props;
+
+  const isDisabled = !!disabled;
+  const initialCheckedState = !!(checked ?? defaultChecked);
+  const [checkedState, setCheckedState] = React.useState(initialCheckedState);
+  const focusRef = isDisabled ? null : componentRef;
+
+  if (defaultChecked !== undefined && checked !== undefined) {
+    console.warn('The props defaultChecked and checked are mutually exclusive. Use only one of the props, do not use both.');
+  }
+
+  const toggleCallback = React.useCallback(
+    (e: InteractionEvent) => {
+      const newCheckedState = checked !== undefined ? checked : !checkedState;
+      onChange && onChange(newCheckedState, e);
+      LayoutAnimation.configureNext(LayoutAnimation.create(200, LayoutAnimation.Types.linear, LayoutAnimation.Properties.scaleX));
```

Comment:
per the controlled vs uncontrolled, we should run the animation off of the state change and not the user interaction

Reply:
Ahh yeah, thanks! I've made it animate off of state change instead.